pkg/addonmanager/addonclients/fluxaddonclient.go

```diff
@@ -274,6 +274,8 @@ func (f *FluxAddonClient) Validations(ctx context.Context, clusterSpec *cluster.
 		return nil
 	}
 
+	clusterSpec.SetDefaultGitOps()
```

Comment:
nit: I wonder if we should move the logic of `SetDefaultGitOps` here or somewhere else (no need to change now, just a thought)

Reply:
yea we can refactor later - for now i added in two validations so we avoid more  SetDefaultGitOps in flux methods